homeassistant/components/hassio/update.py

```diff
@@ -0,0 +1,153 @@
+
+from typing import Any
+
+from awesomeversion import AwesomeVersion
+
+from homeassistant.components.update import IntegrationUpdateFailed, UpdateDescription
+from homeassistant.const import __version__ as CORE_VERSION
+from homeassistant.core import HomeAssistant
+
+from . import get_os_info, get_supervisor_info
+from .const import DOMAIN
+from .handler import HassIO
+
+CORE_VERSION_OBJ = AwesomeVersion(CORE_VERSION)
+
+
+async def async_list_updates(hass: HomeAssistant) -> list[UpdateDescription]:
+    """List all updates available."""
+    updates = []
+    supervisor: HassIO = hass.data[DOMAIN]
+
+    await supervisor.refresh_updates()
+    available_updates = await supervisor.get_available_updates()
+
+    supervisor_info = get_supervisor_info(hass)
+    os_info = get_os_info(hass)
+
+    for update in available_updates["available_updates"]:
+        if update["update_type"] == "core":
+            version_latest = AwesomeVersion(update["version_latest"])
+            if version_latest.dev:
+                changelog = "https://github.com/home-assistant/core/commits/dev"
+            elif version_latest.beta:
+                changelog = "https://rc.home-assistant.io/latest-release-notes/"
+            else:
+                changelog = "https://www.home-assistant.io/latest-release-notes/"
+
+            updates.append(
+                UpdateDescription(
+                    identifier="core",
+                    name="Home Assistant Core",
+                    current_version=CORE_VERSION_OBJ.string,
+                    available_version=version_latest.string,
+                    changelog_url=changelog,
+                    supports_backup=True,
+                )
+            )
+
+        elif update["update_type"] == "supervisor":
+            version_latest = AwesomeVersion(update["version_latest"])
+            if version_latest.dev:
+                changelog = "https://github.com/home-assistant/supervisor/commits/main"
+            else:
+                changelog = f"https://github.com/home-assistant/supervisor/releases/tag/{version_latest.string}"
+
+            updates.append(
+                UpdateDescription(
+                    identifier="supervisor",
+                    name="Home Assistant Supervisor",
+                    current_version=supervisor_info["version"],
+                    available_version=version_latest.string,
+                    changelog_url=changelog,
+                )
+            )
+        elif update["update_type"] == "os":
+            version_latest = AwesomeVersion(update["version_latest"])
+            if version_latest.dev:
+                changelog = (
+                    "https://github.com/home-assistant/operating-system/commits/dev"
+                )
+            else:
+                changelog = f"https://github.com/home-assistant/operating-system/releases/tag/{version_latest.string}"
+
+            updates.append(
+                UpdateDescription(
```

Comment:
Should we add a HA logo ?

Reply:
Or I guess the UI will fallback on the integration brand logo if no icon is set.

Reply:
Yeah, the idea is to have the frontend get from brands repo for the domain when no icon url is defined.